src/vhdl/vhdl-synthesis_checks.adb

```diff
@@ -0,0 +1,165 @@
+--  the Free Software Foundation, either version 2 of the License, or
+--  (at your option) any later version.
+--
+--  This program is distributed in the hope that it will be useful,
+--  but WITHOUT ANY WARRANTY; without even the implied warranty of
+--  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+--  GNU General Public License for more details.
+--
+--  You should have received a copy of the GNU General Public License
+--  along with this program.  If not, see <gnu.org/licenses>.
+with Vhdl.Canon; use Vhdl.Canon;
+with Vhdl.Errors; use Vhdl.Errors;
+with Errorout; use Errorout;
+with Vhdl.Disp_Tree; use Vhdl.Disp_Tree;
+--with Simple_IO; use Simple_IO;
+--with Vhdl.Nodes_Meta; use Vhdl.Nodes_Meta;
+
+
+package body Vhdl.Synthesis_Checks is
+
+   --      "Golden" list of signals is queried by Sensitivity list extraction
+   --      mechanism used in case of "all" keyword. This is a good option,
+   --      because it is LRM standard way of defining "what all should trigger
+   --      the process".
+   function Get_Golden_Sensitivity_List (Proc : Iir) return Iir_List is
+   begin
+      -- FIXME: We have to correctly extract golden list for clocked processes!
+      --        LRM extraction algorithm takes all RHS signals which is not
+      --        desirable in clocked process!
+      return Canon_Extract_Sensitivity_Process (Proc);
+   end Get_Golden_Sensitivity_List;
+
+   -- Checks if each signal from "Golden sensitivity list" are present in
+   -- sensitivity list of a process. Throws a warning for each missing signal.
+   procedure Check_Incomplete_Sensitivity_List (Proc : Iir;
+                                                Gold_List : Iir_List;
+                                                Real_List : Iir_List)
+   is
+      Real_It : List_Iterator;
+      Gold_It : List_Iterator;
+
+      Gold_Iir : Iir;
+      Real_Iir : Iir;
+
+      Match : Boolean;
+   begin
+      Gold_It := List_Iterate (Gold_List);
+      while Is_Valid (Gold_It) loop
+         Gold_Iir := Get_Element (Gold_It);
+         Match := False;
+         Real_It := List_Iterate (Real_List);
+
+         while Is_Valid (Real_It) loop
+            --Put_Line ("Checking pair:");
+            --Put_Line ("Golden:");
+            --Disp_Iir (Get_Element (Gold_It), 4, 1);
+            --Put_Line ("Real:");
+            --Disp_Iir (Get_Element (Real_It), 4, 1);
+            --Put_Line ("Named entity:");
+            --Disp_Iir (Get_Named_Entity (Get_Element (Real_It)), 4, 1);
+            Real_Iir := Get_Element (Real_It);
+            if Get_Named_Entity (Real_Iir) = Gold_Iir then
+               Match := True;
+               exit;
+            end if;
+            Next (Real_It);
+         end loop;
+
+         if Match = False then
+            Warning_Msg_Sem (Warnid_Synth_Sensitivity_Lists, +Proc,
+                             "Incomplete sensitivity list! Missing signal: " &
```

Comment:
You can use a dedicated mechanism to insert names: `... Missing signal: %i", +Gold_Iir);`

Reply:
Ok, will do.